cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -267,3 +268,6 @@ handleSimpleNode p trace nodeTracers nCli nc = do
           then ValidateAllEpochs
           else ValidateMostRecentEpoch
       }
+
+    isProducer :: IsProducer
+    isProducer = error "TODO"
```

Comment:
@Jimbo4350 What's the best way to figure out a node is a producer (= will create blocks) or not?

Reply:
We know this based on whether we provide the leader credentials. We probably don't need the node to tell the consensus this separately, since it already optionally provides the leader credentials. No credentials -> never leader.

Reply:
Unfortunately, this information is hidden somewhere in `Consensus.Protocol blk`. Hmm, maybe it's easier (and less invasive) than I thought. I'll try to cook something up.

Reply:
I pushed a fix.